pkg/api/http.go

```diff
@@ -267,6 +262,13 @@ func ParseSearchBlockRequest(r *http.Request) (*tempopb.SearchBlockRequest, erro
 	}
 	req.Version = version
 
+	s = r.URL.Query().Get(urlParamSize)
+	size, err := strconv.ParseUint(s, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("invalid size %s: %w", s, err)
+	}
+	req.Size_ = size
```

Comment:
Size 0 is still valid for v2 blocks from existing clients (v2 technically doesn't need it since the index has the data page extents, and the index pages themselves can be calculated directly)